src/3d/qgsfgutils.cpp

```diff
@@ -0,0 +1,300 @@
+        fl += formatField( "bufferIndex", QString::number( rs_cast->bufferIndex() ) );
+
+        sl += formatField( "QBlendEquationArguments", formatList( fl ) );
+      }
+      else if ( const auto *rs_cast = qobject_cast<const Qt3DRender::QColorMask *>( rs ) )
+      {
+        QStringList fl;
+        fl += formatField( "red", ( rs_cast->isRedMasked() ? QLatin1String( "true" ) : QLatin1String( "false" ) ) );
+        fl += formatField( "green", ( rs_cast->isGreenMasked() ? QLatin1String( "true" ) : QLatin1String( "false" ) ) );
+        fl += formatField( "blue", ( rs_cast->isBlueMasked() ? QLatin1String( "true" ) : QLatin1String( "false" ) ) );
+        fl += formatField( "alpha", ( rs_cast->isAlphaMasked() ? QLatin1String( "true" ) : QLatin1String( "false" ) ) );
+        sl += formatField( "QColorMask", formatList( fl ) );
+      }
+      else if ( const auto *rs_cast = qobject_cast<const Qt3DRender::QDepthTest *>( rs ) )
+      {
+        sl += formatField( "QDepthTest", QMetaEnum::fromType<Qt3DRender::QDepthTest::DepthFunction>().valueToKey( rs_cast->depthFunction() ) );
+      }
+      else if ( qobject_cast<const Qt3DRender::QNoDepthMask *>( rs ) )
+      {
+        sl += formatField( "QNoDepthMask", "<no_value>" );
+      }
+      else if ( const auto *rs_cast = qobject_cast<const Qt3DRender::QPointSize *>( rs ) )
+      {
+        QStringList fl;
+        fl += formatField( "sizeMode", QMetaEnum::fromType<Qt3DRender::QPointSize::SizeMode>().valueToKey( rs_cast->sizeMode() ) );
+        fl += formatField( "value", QString::number( rs_cast->value() ) );
+        sl += formatField( "QPointSize", formatList( fl ) );
+      }
+      else if ( const auto *rs_cast = qobject_cast<const Qt3DRender::QPolygonOffset *>( rs ) )
+      {
+        QStringList fl;
+        fl += formatField( "scaleFactor", QString::number( rs_cast->scaleFactor() ) );
+        fl += formatField( "depthSteps", QString::number( rs_cast->depthSteps() ) );
+        sl += formatField( "QPolygonOffset", formatList( fl ) );
+      }
+      else if ( qobject_cast<const Qt3DRender::QSeamlessCubemap *>( rs ) )
+      {
+        sl += formatField( "QSeamlessCubemap", "<no_value>" );
+      }
+    }
+    res += QString( " %1" ).arg( formatList( sl ) );
+  }
+
+  else if ( const auto *rs = qobject_cast<const Qt3DRender::QRenderTargetSelector *>( n ) )
+  {
+    if ( rs->target() )
+    {
+      QStringList sl;
+      const auto outputs = rs->target()->outputs();
+      for ( auto o : outputs )
+      {
+        sl += formatField( QMetaEnum::fromType<Qt3DRender::QRenderTargetOutput::AttachmentPoint>().valueToKey( o->attachmentPoint() ),
+                           formatIdName( o->texture() ) );
+      }
+      QStringList fl;
+      fl +=  formatField( QLatin1String( "outputs" ),
+                          formatList( sl ) );
+      res += QString( " %1" ).arg( formatList( fl ) );
+    }
+  }
+  //    if (!n->isEnabled())
+  //        res += QLatin1String(" [D]");
```

Comment:
why is it commented?

Reply:
dead code... see `QgsFgUtils::formatNode`